content/blog/2020/07/2020-07-27-machine-learning-plugin-coding-phase2.adoc

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+title: "Machine Learning Plugin project - Coding Phase 2 blog post"
+tags:
+- machinelearning
+- datascience
+- plugins
+- communitybonding
+- gsoc
+- gsoc2020
+author: loghijiaha
+description: "Coding Phase 2 of Google Summer of Code 2020"
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+image::/images/gsoc/jenkins-gsoc-logo_small.png[align="center"]
+
+Welcome back folks!
+
+This blog post is about my coding phase 2 in Jenkins link:/projects/gsoc/2020/projects/machine-learning/[Machine Learning Plugin] for this GSoC 2020.
+After successfully passing the evaluation and demo in the phase 1, our team went ahead for facing the challenges in phase 2.
+
+== Summary
+
+This phase of coding was well spent by documentation and by fixing many bugs.
+As the main feature of connecting to an IPython Kernel is done in phase 1, we were able to focus on fixing minor/major bugs and documenting for the users.
+According to the link:https://issues.jenkins-ci.org/browse/JENKINS-62927[JENKINS-62927] issue, a Docker agent was built to facilitate users without concerning plugin dependencies in python.
+In the act of deprecation of Python 2, we ported our plugin to support Python 3.
+We have tested our plugin in conda, venv and windows environments.
+Machine learning plugin has successfully passed the end to end test. A feature for a code editor is needed for further discussion/analysis as we have done a simple editor that may be useful in other ways in the future. link:https://github.com/jenkinsci/machine-learning-plugin/pull/35[PR#35]
+
+== Main features of Machine Learning plugin
+
+- Run Jupyter notebook, (Zeppelin) JSON and Python files
+- Run Python code directly
+- Convert Jupyter Notebooks to Python and JSON
+- Configure IPython kernel properties
+- Support to execute Notebooks/Python on Agent
+- Support for Windows and Linux
+
+== Upcoming features
+
+- Extract graph/map/images from the code
+- Save artifacts according to the step name
+- Generate reports for corresponding build
+
+== Future improvements
+
```

Comment:
Do you plan to do it during GSoC? If not, maybe it makes sense to say it explicitly to highlight there is no committment.